src/main/java/de/tum/in/www1/artemis/repository/ParticipationRepository.java

```diff
@@ -16,11 +16,14 @@
 @Repository
 public interface ParticipationRepository extends JpaRepository<Participation, Long> {
 
-    List<Participation> findByExerciseId(@Param("exerciseId") Long exerciseId);
-
     @Query("select p from Participation p where p.exercise.course.id = :courseId")
     List<Participation> findByCourseId(@Param("courseId") Long courseId);
 
+    @Query("select participation from Participation participation where participation.exercise.course.id = :courseId and participation.student.login= :username")
+    List<Participation> findByCourseIdAndStudentLogin(@Param("courseId") Long courseId, @Param("username") String username);
+
+    List<Participation> findByExerciseId(@Param("exerciseId") Long exerciseId);
+
     Participation findOneByExerciseIdAndStudentLogin(Long exerciseId, String username);
```

Comment:
Can you revert these changes as well, if they are not needed?

Reply:
Done :)